### PR TITLE
test: K8sVerifier Fix test-verifier's scheduling

### LINKED ISSUE #14791: CI: K8sVerifier Runs the kernel verifier against Cilium's BPF datapath: FailedScheduling
https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/1535/testReport/Suite-k8s-1/11/K8sVerifier_Runs_the_kernel_verifier_against_Cilium_s_BPF_datapath/
https://datastudio.google.com/s/l_YypLc1mUc

I've only seen this failure happen on v1.8 with K8s 1.11 so it could be something specific to K8s 1.11.

### Stacktrace
```
/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:461
test-verifier pod not ready after timeout
Expected
    <*errors.errorString | 0xc00047a640>: {
        s: "timed out waiting for pods with filter test-verifier to be ready: 4m0s timeout expired",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test/k8sT/Verifier.go:58
```
### Standard Error
```
21:31:37 STEP: Running BeforeAll block for EntireTestsuite
21:31:37 STEP: Starting tests: command line parameters: {Reprovision:false HoldEnvironment:false PassCLIEnvironment:true SSHConfig: ShowCommands:false TestScope: SkipLogGathering:false CiliumImage: CiliumOperatorImage: HubbleRelayImage: ProvisionK8s:true Timeout:5h0m0s Kubeconfig:/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test/vagrant-kubeconfig Registry:147.75.39.181 RegistryCredentials: Benchmarks:false Multinode:true RunQuarantined:false Help:false} environment variables: [JENKINS_HOME=/var/jenkins_home ghprbSourceBranch=pr/v1.8-backport-2021-01-28 ghprbTriggerAuthorEmail=joestringernz@gmail.com VM_MEMORY=8192 MAIL=/var/mail/root KUBEPROXY=0 SSH_CLIENT=54.148.123.155 45332 22 ghprbPullAuthorEmail=joestringernz@gmail.com USER=root PROJ_PATH=src/github.com/cilium/cilium RUN_CHANGES_DISPLAY_URL=https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/1535/display/redirect?page=changes ghprbPullDescription=GitHub pull request #14782 of commit 7458c18daf73e30a2149e0de692fb7c3c5f9bf11, no merge conflicts. NETNEXT=1 ghprbActualCommit=7458c18daf73e30a2149e0de692fb7c3c5f9bf11 SHLVL=1 NODE_LABELS=baremetal ginkgo nightly node-proud-baboon vagrant HUDSON_URL=https://jenkins.cilium.io/ GIT_COMMIT=f170caa0881152c856da10e160d46382f65a79f8 OLDPWD=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next GINKGO_TIMEOUT=300m HOME=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next ghprbTriggerAuthorLoginMention=@joestringer BUILD_URL=https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/1535/ ghprbPullAuthorLoginMention=@joestringer HUDSON_COOKIE=cbffb796-b5cc-4c48-8689-56fa3e202fb5 JENKINS_SERVER_COOKIE=durable-e347915103f4a5a594b55c3295847857 ghprbGhRepository=cilium/cilium JobKernelVersion=net-next KERNEL=net-next CONTAINER_RUNTIME=docker COMMENT_BODY=test-backport-1.8 WORKSPACE=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next ghprbPullLongDescription=* #14730 -- bpf: Send packet drop notify for ipv6 lb nat mode failures. (@hzhou8)\n * #14707 -- contrib: Add script to fetch docker manifests (@joestringer)\n\nOnce this PR is merged, you can update the PR labels via:\n```upstream-prs\n$ for pr in 14730 14707; do contrib/backporting/set-labels.py $pr done 1.8; done\n``` K8S_NODES=3 TESTDIR=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test LOGNAME=root NODE_NAME=node-proud-baboon ghprbCredentialsId=ciliumbot _=/usr/bin/java STAGE_NAME=BDD-Test-PR GIT_BRANCH=origin/pr/14782/merge EXECUTOR_NUMBER=0 NO_CILIUM_ON_NODE=k8s3 ghprbTriggerAuthorLogin=joestringer TERM=xterm XDG_SESSION_ID=5 HOST_FIREWALL=0 BUILD_DISPLAY_NAME=v1.8 backports 2021-01-28  https://github.com/cilium/cilium/pull/14782  #1535 ghprbPullAuthorLogin=joestringer HUDSON_HOME=/var/jenkins_home ghprbTriggerAuthor=Joe Stringer JOB_BASE_NAME=Cilium-PR-K8s-1.11-net-next PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/go/bin:/root/go/bin sha1=origin/pr/14782/merge KUBECONFIG=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test/vagrant-kubeconfig FOCUS=K8s BUILD_ID=1535 XDG_RUNTIME_DIR=/run/user/0 BUILD_TAG=jenkins-Cilium-PR-K8s-1.11-net-next-1535 RUN_QUARANTINED=false JENKINS_URL=https://jenkins.cilium.io/ LANG=en_US.UTF-8 ghprbCommentBody=test-backport-1.8 JOB_URL=https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/ ghprbPullTitle=v1.8 backports 2021-01-28 GIT_URL=https://github.com/cilium/cilium ghprbPullLink=https://github.com/cilium/cilium/pull/14782 BUILD_NUMBER=1535 JENKINS_NODE_COOKIE=57882438-f0ef-49b6-b193-2dcaa8d58294 SHELL=/bin/bash GOPATH=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next RUN_DISPLAY_URL=https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/1535/display/redirect ghprbAuthorRepoGitUrl=https://github.com/cilium/cilium.git FAILFAST=false HUDSON_SERVER_COOKIE=693c250bfb7e85bf ghprbTargetBranch=v1.8 JOB_DISPLAY_URL=https://jenkins.cilium.io/job/Cilium-PR-K8s-1.11-net-next/display/redirect K8S_VERSION=1.11 JOB_NAME=Cilium-PR-K8s-1.11-net-next PWD=/home/jenkins/workspace/Cilium-PR-K8s-1.11-net-next/src/github.com/cilium/cilium/test SSH_CONNECTION=54.148.123.155 45332 147.75.39.181 22 ghprbPullId=14782 JobK8sVersion=1.11 VM_CPUS=3 CILIUM_REGISTRY=147.75.39.181]
21:31:37 STEP: Ensuring the namespace kube-system exists
21:31:37 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs")
21:31:42 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs") => <nil>
21:31:42 STEP: Preparing cluster
21:31:42 STEP: Labelling nodes
21:31:42 STEP: Running BeforeAll block for EntireTestsuite K8sVerifier
21:31:42 STEP: Ensuring the namespace kube-system exists
21:31:43 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs")
21:31:43 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs") => <nil>
21:31:43 STEP: WaitforPods(namespace="default", filter="test-verifier")
21:35:43 STEP: WaitforPods(namespace="default", filter="test-verifier") => timed out waiting for pods with filter test-verifier to be ready: 4m0s timeout expired
21:35:43 STEP: cmd: kubectl describe pods -n default test-verifier
Exitcode: 0 
Stdout:
 	 Name:         test-verifier
	 Namespace:    default
	 Priority:     0
	 Node:         <none>
	 Labels:       <none>
	 Annotations:  <none>
	 Status:       Pending
	 IP:           
	 IPs:          <none>
	 Containers:
	   cilium-builder:
	     Image:      quay.io/cilium/cilium-builder:2021-01-20-v1.8@sha256:545a7837bc135a6f3193c791a0569a2897cf75a418b27e996a51c451cd245df6
	     Port:       <none>
	     Host Port:  <none>
	     Command:
	       sleep
	     Args:
	       1000h
	     Environment:  <none>
	     Mounts:
	       /cilium from cilium-src (rw)
	       /sys/fs/bpf from bpf-maps (rw)
	       /var/run/secrets/kubernetes.io/serviceaccount from default-token-vvm44 (ro)
	 Conditions:
	   Type           Status
	   PodScheduled   False 
	 Volumes:
	   bpf-maps:
	     Type:          HostPath (bare host directory volume)
	     Path:          /sys/fs/bpf
	     HostPathType:  DirectoryOrCreate
	   cilium-src:
	     Type:          HostPath (bare host directory volume)
	     Path:          /home/vagrant/go/src/github.com/cilium/cilium
	     HostPathType:  Directory
	   default-token-vvm44:
	     Type:        Secret (a volume populated by a Secret)
	     SecretName:  default-token-vvm44
	     Optional:    false
	 QoS Class:       BestEffort
	 Node-Selectors:  <none>
	 Tolerations:     node.kubernetes.io/not-ready op=Exists
	                  node.kubernetes.io/unreachable op=Exists
	 Events:
	   Type     Reason            Age               From               Message
	   ----     ------            ----              ----               -------
	   Warning  FailedScheduling  36s (x8 over 4m)  default-scheduler  0/3 nodes are available: 3 node(s) were not ready.
	 
Stderr:
 	 

FAIL: test-verifier pod not ready after timeout
Expected
    <*errors.errorString | 0xc00047a640>: {
        s: "timed out waiting for pods with filter test-verifier to be ready: 4m0s timeout expired",
    }
to be nil
===================== TEST FAILED =====================
21:35:43 STEP: Running AfterFailed block for EntireTestsuite K8sVerifier
Name:         test-verifier
Namespace:    default
Priority:     0
Node:         <none>
Labels:       <none>
Annotations:  <none>
Status:       Pending
IP:           
IPs:          <none>
Containers:
  cilium-builder:
    Image:      quay.io/cilium/cilium-builder:2021-01-20-v1.8@sha256:545a7837bc135a6f3193c791a0569a2897cf75a418b27e996a51c451cd245df6
    Port:       <none>
    Host Port:  <none>
    Command:
      sleep
    Args:
      1000h
    Environment:  <none>
    Mounts:
      /cilium from cilium-src (rw)
      /sys/fs/bpf from bpf-maps (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-vvm44 (ro)
Conditions:
  Type           Status
  PodScheduled   False 
Volumes:
  bpf-maps:
    Type:          HostPath (bare host directory volume)
    Path:          /sys/fs/bpf
    HostPathType:  DirectoryOrCreate
  cilium-src:
    Type:          HostPath (bare host directory volume)
    Path:          /home/vagrant/go/src/github.com/cilium/cilium
    HostPathType:  Directory
  default-token-vvm44:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-vvm44
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready op=Exists
                 node.kubernetes.io/unreachable op=Exists
Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  36s (x8 over 4m)  default-scheduler  0/3 nodes are available: 3 node(s) were not ready.

===================== Exiting AfterFailed =====================
21:35:43 STEP: Running AfterEach for block EntireTestsuite
```

### PR DESCRIPTION
**Pull request by pchaigno**
See commits for details.

Fixes: https://github.com/cilium/cilium/issues/14791

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

File: test/k8sT/manifests/test-verifier.yaml
Comment (aanm) [inline test/k8sT/manifests/test-verifier.yaml]: None of the manifests have the node name specified. Would it make sense to think on a solution that doesn't have infrastructure details in the tests?

Comment (nebril) [reply test/k8sT/manifests/test-verifier.yaml]: At the very least we should use `cilium.io/ci=k8s1` label

Comment (pchaigno) [reply test/k8sT/manifests/test-verifier.yaml]: We already indirectly encode an infrastructure detail in the test. We currently need to mount the Cilium sources inside the `test-verifier` container. That means we can only run this test when the sources are available inside the VM (no GKE).

And I just found out that the second flake affecting `test-verifier` (cf. https://github.com/cilium/cilium/issues/14792) only happens when the pod is scheduled on k8s2. If it wasn't for that second flake, we could use a DaemonSet I guess.

**Comment by pchaigno** [comment]
test-4.9

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-netnext

**Comment by pchaigno** [comment]
test-netnext

**Comment by pchaigno** [comment]
K8s-1.13-Kernel-netnext hit known flake https://github.com/cilium/cilium/issues/12511. Other builds are green.

**Review by aanm — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Comment by pchaigno** [comment]
Although this test is specific to @cilium/bpf, the change implemented in this PR has nothing to do with our datapath or BPF, so I don't think we need to wait for a review from @cilium/bpf. Only test failure is a flake as indicated above. Marking as ready to merge.

**Review by tklauser — APPROVED** [review]

File: test/k8sT/manifests/test-verifier.yaml
Comment (pchaigno) [reply test/k8sT/manifests/test-verifier.yaml]: Unfortunately switching to the use of labels to match the node broke the fix for the scheduling issue. See https://github.com/cilium/cilium/issues/14791#issuecomment-788176744. https://github.com/cilium/cilium/pull/15156 should have the new fix, needed only on v1.8.

**Pull request merged**
